packages/apollo-link-ws/src/webSocketLink.ts

```diff
@@ -23,11 +30,51 @@ export class WebSocketLink extends ApolloLink {
         paramsOrClient.webSocketImpl,
       );
     }
+    this.isRequeried = options.isRequeried;
+    if (options.connectionCallback) {
```

Comment:
What if we provided a generic subscriber on the link itself instead of like this?

```js
const wsLink = new WSLink();

wsLink.on('connected', () => {})
```

Reply:
@jbaxleyiii you can kind of do this already, by hooking up on the connected event on the ws client directly

```
const client = new SubscriptionClient(...);
client.onConnected(() => console.log('connected'));
```

Reply:
@dozoisch - true but I guess there is the case where you pass params rather than a client...

Reply:
@jbaxleyiii ok done that..